### DISCUSSION TITLE
Creating and Signing a TON Transaction with Trust Wallet Core

### DISCUSSION THREAD

**Discussion by ronaldoguedess** [trustwallet/wallet-core#3870] [Q&A] [answered]
I've been exploring Trust Wallet and noticed its capability to generate TON addresses. I'm now interested in building a transaction and signing it using Trust Wallet Core.

However, I couldn't find any relevant examples in the documentation: [Trust Wallet Core Android Test](https://github.com/trustwallet/wallet-core/tree/master/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains).

Does anyone have a sample or guidance on how to achieve this? Any help would be greatly appreciated!

**Comment by satoshiotomakan** [comment, marked as answer]
Hi @ronaldoguedess, isn't this what you need?
https://github.com/trustwallet/wallet-core/blob/master/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/theopennetwork/TestTheOpenNetworkAddress.kt
